addon/adapters/rest.js

```diff
@@ -1454,6 +1405,50 @@ if (isEnabled('ds-improved-ajax')) {
 
 }
 
+function ajaxSuccess(adapter, jqXHR, payload, requestData, resolve, reject) {
```

Comment:
lets not pass reject/resolve here, instead simply `return` or `throw` (or `return Promise.reject()`)